server/auth/sso/sso.go

```diff
@@ -274,7 +274,7 @@ func (s *sso) HandleCallback(w http.ResponseWriter, r *http.Request) {
 		Secure:   s.secure,
 	})
 	redirect := s.baseHRef
-	if cookie.Value != "" {
+	if strings.HasPrefix(cookie.Value, s.baseHRef) {
```

Comment:
test?

Reply:
Currently the following redirects to Google after SSO:

https://argo/login?redirect=https://google.com

With the fix, it ignores the redirect cookie (set to https://google.com) and it redirects back to https://argo.

Since the auto-generated redirect URL always has the baseHRef, this enforces that and prevents manipulation.

Reply:
This condition is failing for all redirect URLs.  Need to add the proto and host.

prefix := fmt.Sprintf("%s://%s%s", proto, r.Host, s.baseHRef)